File: backend/app/services/assignment_service.py
from sqlalchemy.orm import Session

from app.repositories.assignment_repository import AssignmentRepository
from app.schemas.assignment import (
    AssignmentCreate,
    AssignmentUpdate,
)


class AssignmentService:

    @staticmethod
    def get_all(db: Session):
        return AssignmentRepository.get_all(db)

    @staticmethod
    def get_by_id(
        db: Session,
        assignment_id: int,
    ):
        return AssignmentRepository.get_by_id(
            db,
            assignment_id,
        )

    @staticmethod
    def create(
        db: Session,
        assignment: AssignmentCreate,
    ):
        return AssignmentRepository.create(
            db,
            assignment,
        )

    @staticmethod
    def update(
        db: Session,
        assignment_id: int,
        assignment: AssignmentUpdate,
    ):
        return AssignmentRepository.update(
            db,
            assignment_id,
            assignment,
        )

    @staticmethod
    def delete(
        db: Session,
        assignment_id: int,
    ):
        return AssignmentRepository.delete(
            db,
            assignment_id,
        )

File: frontend/features/assignment-management/services/assignment.service.ts
import { AssignmentApi } from "../api/assignmentApi";

export const AssignmentService = {
  getAll: () => AssignmentApi.getAll(),

  getById: (id: number) =>
    AssignmentApi.getById(id),

  create: (payload: any) =>
    AssignmentApi.create(payload),

  update: (id: number, payload: any) =>
    AssignmentApi.update(id, payload),

  remove: (id: number) =>
    AssignmentApi.remove(id),
};